File: Zisimopoulos_Valencia/Software Construction/QL/src/uva/sc/logic/Literal.java
package uva.sc.logic;


public class Literal extends Node{
	String str;
	Double d;
	int i;
	boolean bool;
	
	public Literal (String str){
		this.str = str;
	}
	
	public Literal (Double d){
		this.d = d;
	}
	
	public Literal (int i){
		this.i = i;
	}
	
	public Literal (boolean bool){
		this.bool = bool;
	}
	
	public String toString() {
		return "[String]: " + this.str + "[Double]: " + this.d + 
			   "[Integer]: " + this.i + "[Boolean]: " + this.bool;
	}
	
}

File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/Literals/Literal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Model.Statements;
using UvA.SoftCon.Questionnaire.AST.Types;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions.Literals
{
    public abstract class Literal<T> : Node, IExpression
    {
        public T Value
        {
            get;
            private set;
        }

        protected Literal(T value, TextPosition position)
            : base(position)
        {
            Value = value;
        }

        public abstract DataType? GetType(IDictionary<string, DataType> symbolTable);

        public abstract IValue Evaluate(IDictionary<string, IValue> environment);

        public override string ToString()
        {
            return Value.ToString();
        }
    }
}
